src/components/single-line-diagram.js

```diff
@@ -183,7 +187,117 @@ const SingleLineDiagram = forwardRef((props, ref) => {
     };
 
     useLayoutEffect(() => {
+        function addNavigationArrowDef() {
+            let svg = document.getElementById('sld-svg').children[0]; //Get svg element
+            let defElement = document.createElementNS(
+                'http://www.w3.org/2000/svg',
+                'defs'
+            );
+
+            let html;
+            if (theme === LIGHT_THEME) {
+                html = [
+                    '<marker id="arrowhead" viewBox="0 0 100 100" refX="50" refY="1" >',
+                    '<polygon style="stroke: white;fill: white" points="0 0, 50 80, 100 0"/>',
+                    '</marker>',
+                ].join('');
+            } else {
+                html = [
+                    '<marker id="arrowhead" viewBox="0 0 100 100" refX="50" refY="1" >',
+                    '<polygon style="stroke: #212121;fill: #212121" points="0 0, 50 80, 100 0"/>',
+                    '</marker>',
+                ].join('');
+            }
+
+            defElement.innerHTML = html;
+            svg.appendChild(defElement);
+        }
+
+        function createSvgArrow(element, transform, position) {
```

Comment:
Why do you generate the svg in js? Why not an external SVG resource that will be added to decorate the original one? This is a lot of boilerplate code that we don't want to see.